core/src/server/handlers/auth/auth_params_parsing_test.cpp

```diff
@@ -0,0 +1,185 @@
+#include <userver/server/handlers/auth/auth_params_parsing.hpp>
+#include <userver/utest/utest.hpp>
+
+#include <gtest/gtest.h>
+
+#include <exception>
+#include <string_view>
+
+USERVER_NAMESPACE_BEGIN
+
+TEST(AuthenticationInfoCorrectParsing, WithoutOptional) {
+    std::string_view correctInfo = R"(username="Mufasa",
+        realm="testrealm@host.com",
+        nonce="dcd98b7102dd2f0e8b11d0f600bfb0c093",
+        uri="/dir/index.html",
+        response="6629fae49393a05397450978507c4ef1"
+    )";
+    userver::server::handlers::auth::DigestParsing parser;
+    EXPECT_NO_THROW(parser.ParseAuthInfo(correctInfo));
+    auto auth_context = parser.GetClientContext();
+
+    EXPECT_EQ(auth_context.username, "Mufasa");
+    EXPECT_EQ(auth_context.nonce, "dcd98b7102dd2f0e8b11d0f600bfb0c093");
+    EXPECT_EQ(auth_context.realm, "testrealm@host.com");
+    EXPECT_EQ(auth_context.uri, "/dir/index.html");
+    EXPECT_EQ(auth_context.response, "6629fae49393a05397450978507c4ef1");
+}
+
+TEST(AuthenticationInfo, WithOptionalPartial) {
+    std::string_view correctInfo = R"(username="Mufasa",
+        realm="testrealm@host.com",
+        nonce="dcd98b7102dd2f0e8b11d0f600bfb0c093",
+        uri="/dir/index.html",
+        response="6629fae49393a05397450978507c4ef1",
+        algorithm="MD5"
```

Comment:
For historical reasons, a sender MUST NOT generate the quoted string
   syntax for the following parameters: algorithm, qop, and nc.